pkg/controller/eventinstance/eventinstance.go

```diff
@@ -0,0 +1,105 @@
+package eventinstance
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+	"golang.org/x/sync/semaphore"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const defaultTTL = 168 * time.Hour // 7 days
+
+type Handler struct {
+	sem    *semaphore.Weighted
+	rawTTL string
+	ttl    time.Duration
+}
+
+func NewHandler() *Handler {
+	return &Handler{
+		sem: semaphore.NewWeighted(1),
+	}
+}
+
+func (h *Handler) UpdateTTL(req router.Request, _ router.Response) error {
+	if err := h.sem.Acquire(req.Ctx, 1); err != nil {
```

Comment:
I switched to a semaphore because it lets us set a deadline using a context (in this case, it shares a deadline with the request)